cedar-policy-generators/src/schema.rs

```diff
@@ -820,49 +816,34 @@ impl Schema {
                 selected_indices.push(indices.swap_remove(index));
             }
 
-            Result::Ok(Some(
+            Result::Ok(
                 selected_indices
                     .iter()
                     .map(|&i| {
                         let (name, _) = &entity_types[i];
-                        tys.insert(build_qualified_entity_type_name(
-                            namespace.as_ref(),
-                            &name.clone().into(),
-                        ));
-                        name.clone().into()
+                        let etyp: ast::EntityType = ast::Name::from(name.clone()).into();
+                        tys.insert(build_qualified_entity_type_name(namespace.as_ref(), &etyp));
+                        etyp
                     })
-                    .collect::<Vec<Name>>(),
-            ))
+                    .collect::<Vec<ast::EntityType>>(),
+            )
         };
-        let mut principal_and_resource_types_exist = false;
         let mut actions: Vec<(SmolStr, ActionType)> = action_names
             .iter()
             .map(|name| {
                 Ok((
                     name.clone(),
                     ActionType {
                         applies_to: {
-                            let mut picked_resource_types =
-                                pick_entity_types(&mut resource_types, u)?;
-                            let mut picked_principal_types =
+                            let picked_resource_types = pick_entity_types(&mut resource_types, u)?;
+                            let picked_principal_types =
                                 pick_entity_types(&mut principal_types, u)?;
                             // If we already have resource_types and principal_types, randomly make them empty
-                            if principal_and_resource_types_exist {
-                                if u.ratio(1, 8)? {
-                                    picked_resource_types = None;
-                                }
-                                if u.ratio(1, 8)? {
-                                    picked_principal_types = None;
-                                }
-                            } else {
-                                principal_and_resource_types_exist = true;
-                            }
```

Comment:
Remove comment since you're removing this code. But also: why remove this code?

Reply:
`principal` and `resource` can't be `None` anymore